build/source/engine/allocspace.f90

```diff
@@ -19,405 +19,348 @@
+ integer(i4b),            intent(in)   :: nHRU          ! number of HRUs
+ integer(i4b),            intent(in)   :: nSnow(:)      ! number of snow layers for each HRU
+ integer(i4b),            intent(in)   :: nSoil(:)      ! number of soil layers for each HRU
+ ! output: data structures
+ type(var_i),             intent(out)  :: timeStruct    ! model time data
+ type(spatial_double),    intent(out)  :: forcStruct    ! model forcing data
+ type(spatial_double),    intent(out)  :: attrStruct    ! local attributes for each HRU
+ type(spatial_int),       intent(out)  :: typeStruct    ! local classification of soil veg etc. for each HRU
+ type(spatial_double),    intent(out)  :: mparStruct    ! model parameters
+ type(spatial_intVec),    intent(out)  :: indxStruct    ! model indices
+ type(spatial_doubleVec), intent(out)  :: progStruct    ! model prognostic (state) variables
+ type(spatial_doubleVec), intent(out)  :: diagStruct    ! model diagnostic variables
+ type(spatial_doubleVec), intent(out)  :: fluxStruct    ! model fluxes
+ type(var_dlength),       intent(out)  :: derivStruct   ! model derivatives
+ type(var_d),             intent(out)  :: bparStruct    ! basin-average parameters
+ type(var_dlength),       intent(out)  :: bvarStruct    ! basin-average variables
+ ! output: error control
+ integer(i4b),intent(out)              :: err           ! error code
+ character(*),intent(out)              :: message       ! error message
+ ! ---------------------------------------------------------------------------------------------------------------
+ ! local variables
+ integer(i4b)                          :: iStruct       ! index of data structure 
+ character(len=256)                    :: cmessage      ! error message of downwind routine
  ! initialize errors
- err=0; message="init_model/"
- ! ensure metadata structures are deallocated
- if (associated(time_meta)) deallocate(time_meta)
- if (associated(forc_meta)) deallocate(forc_meta)
- if (associated(attr_meta)) deallocate(attr_meta)
- if (associated(type_meta)) deallocate(type_meta)
- if (associated(mpar_meta)) deallocate(mpar_meta)
- if (associated(mvar_meta)) deallocate(mvar_meta)
- if (associated(indx_meta)) deallocate(indx_meta)
- if (associated(bpar_meta)) deallocate(bpar_meta)
- if (associated(bvar_meta)) deallocate(bvar_meta)
- ! allocate metadata structures
- allocate(time_meta(maxvarTime),forc_meta(maxvarForc),attr_meta(maxvarAttr),type_meta(maxvarType),&
-          mpar_meta(maxvarMpar),mvar_meta(maxvarMvar),indx_meta(maxvarIndx),&
-          bpar_meta(maxvarBpar),bvar_meta(maxvarBvar),stat=err)
- if(err/=0)then; err=20; message=trim(message)//"problemAllocateMetadata"; return; endif
- end subroutine init_metad
+ err=0; message="initStruct/"
 
+ ! loop through data structures
+ do iStruct=1,size(structInfo)
+  ! allocate space
+  select case(trim(structInfo(iStruct)%structName))
+   case('time'); call allocGlobal(nHRU, time_meta,  timeStruct,  err, cmessage)   ! model forcing data
```

Comment:
@martynpclark : are you planning to update the spacing before this is merged? If not, then I will just open an issue for this.